src/table/hooks/use-column-widths.ts

```diff
@@ -79,14 +79,18 @@ export const getColumnWidths = (
 
 const useColumnWidths = (
   columns: Column[],
-  tableWidth: number
+  tableWidth: number,
+  { head, body }: TableStyling
 ): [number[], React.Dispatch<React.SetStateAction<number[]>>] => {
-  const measureTextMethods = useMeasureText('13px', 'Arial');
-  // TODO use actual font size (and font eventually)
-  const [columnWidths, setColumnWidths] = useState(getColumnWidths(columns, tableWidth, measureTextMethods));
+  const measureHeadLabel = useMeasureText(head.fontSize, head.fontFamily).measureText;
+  const { measureText, estimateWidth } = useMeasureText(body.fontSize, body.fontFamily);
+  const getHugWidth = (headLabel: string, totalsLabel: string, glyphCount: number) =>
+    Math.max(measureHeadLabel(headLabel), measureText(totalsLabel), estimateWidth(glyphCount));
+
+  const [columnWidths, setColumnWidths] = useState(getColumnWidths(columns, tableWidth, getHugWidth));
```

Comment:
Try to avoid calling `getColumnWidths` every time this hook runs. As it can become costly to run that function everytime the state is updated just to discard the "initial value".

Image a table with 2000 column. Running `getColumnWidths` every state update.

Reply:
```suggestion
  const [columnWidths, setColumnWidths] = useState(() => getColumnWidths(columns, tableWidth, getHugWidth));
```

Reply:
Oh I didn't know that, good to know